### PR TITLE
Updating gRPC templates

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Making the gRPC templates a single project template by removing the client. I have manually tested the template on CLI and VS as well as design time build in VS. This PR also supersede the version update PR: https://github.com/aspnet/AspNetCore/pull/9047

WIP:
- [x] Fix the tests @ryanbrandenburg added
- [x] ~~Explore options to make design time build work without extra junk in the csproj.~~ Will do separately.
- [x] FWLink and doc for client

### CODE REVIEW

File: eng/scripts/CodeCheck.ps1 [hunk 1]
```diff
@@ -89,7 +89,7 @@ try {
     Get-ChildItem "$repoRoot/*.sln" -Recurse `
         | ? {
             # These .sln files are used by the templating engine.
-            ($_.Name -ne "RazorComponentsWeb-CSharp.sln") -and ($_.Name -ne "GrpcService-CSharp.sln")
```
Comment (JamesNK) [inline eng/scripts/CodeCheck.ps1]: Why is this removed?

**Review by JamesNK — COMMENTED** [review]
Also need to add instructions for creating a client so the new service is usable.

Comments + link to the server project somewhere

File: eng/scripts/CodeCheck.ps1 (hunk 1 above)
Comment (JunTaoLuo) [reply eng/scripts/CodeCheck.ps1]: This was only required for multi-project templates that include a sln. This is now removed.

**Comment by JunTaoLuo** [comment]
@shirhatti @JamesNK On the topic of design time builds:
Good news: It's possible to remove the extra itemgroups operations (content and none) and require only `<Protobuf Include="Protos\*.proto" GrpcServices="Server" />` in the .csproj.
Bad news: It's going to require a small update to either `Google.Protobuf.Tools.targets` or `_Grpc.Tools.targets` in Grpc.Tools. I'm not sure if we can get it in for 1.20 at this point.

**Comment by JamesNK** [comment]
There won't be a Google.Protobuf release anytime soon. If the change is to Grpc.* then we might be able to sneak it in. Do a PR for it today to get the ball rolling.

**Comment by JunTaoLuo** [comment]
As I was verifying the Design Time Build scenarios, it turns out that the way we enable it (single file generator, adding to content and removing from none itemgroups) may have side effects with other VS gestures such as adding files. I'm in conversation with vs tooling to confirm if there are any issues. But due to this uncertainty, I'd rather keep that in the templates for now instead of production targets in Grpc.Tools. Is that okay?

**Comment by shirhatti** [comment]
@davidfowl @DamianEdwards We're changing the gRPC template to a single project template as per our discussion. There is no longer a client project. We've added a FWLink as a code which will point to a doc on how to generate a client.

**Comment by shirhatti** [comment]
> But due to this uncertainty, I'd rather keep that in the templates for now instead of production targets in Grpc.Tools. Is that okay?

Sure. Let's work with tooling folks to get a better understanding of that are the impacted VS gestures.

File: src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Startup.cs [hunk 2]
```diff
@@ -30,6 +30,8 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
             app.UseEndpoints(endpoints =>
             {
+                // Communication with gRPC endpoints must be made through a gRPC client.
+                // To learn how to create a client, visit: https://go.microsoft.com/fwlink/?linkid=2086909.
```
Comment (JamesNK) [inline src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Startup.cs]: trim the trailing fullstop

Comment (JamesNK) [inline src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Startup.cs]:
```suggestion
                // To learn how to create a client, visit: https://go.microsoft.com/fwlink/?linkid=2086909
```

**Review by ryanbrandenburg — APPROVED** [review]

**Review by JamesNK — APPROVED** [review]

**Review by shirhatti — APPROVED** [review]

**Comment by JunTaoLuo** [comment]
Hmm @dougbu  @aspnet/brt Hey have you guys seem failures like:

```
F:\workspace\_work\1\s\.dotnet\buildtools\korebuild\3.0.0-build-20190322.1\modules\vstest\module.targets(158,5): error : Test group UnitTests/netcoreapp3.0 failed with exit code '1'. [F:\workspace\_work\1\s\.dotnet\buildtools\korebuild\3.0.0-build-20190322.1\KoreBuild.proj]
```

```
  fail: Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler[12]
        Message contains error: 'access_denied', error_description: 'AADB2C90118', error_uri: 'error_uri is null'.
  info: Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler[4]
        Error from RemoteAuthentication: Message contains error: 'access_denied', error_description: 'AADB2C90118', error_uri: 'error_uri is null'..
  info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
        Request finished in 44.9797ms 302 
  WARNING: Could not set default HttpMessageHandler because 'GetHttpMessageHandler' was not found on 'System.Net.Http.HttpClient'.
  WARNING: Could not set default HttpMessageHandler because 'GetHttpMessageHandler' was not found on 'System.Net.Http.HttpClient'.
  [xUnit.net 00:00:06.64]     Microsoft.AspNetCore.Blazor.Build.Test.ComponentRenderingRazorIntegrationTest.Render_HtmlBlock_Integration [SKIP]
  Skipped  Microsoft.AspNetCore.Blazor.Build.Test.ComponentRenderingRazorIntegrationTest.Render_HtmlBlock_Integration
  [xUnit.net 00:00:06.88]     Microsoft.AspNetCore.Blazor.Build.Test.ComponentRenderingRazorIntegrationTest.Render_Component_HtmlEncoded [SKIP]
  Skipped  Microsoft.AspNetCore.Blazor.Build.Test.ComponentRenderingRazorIntegrationTest.Render_Component_HtmlEncoded
  Results File: F:\workspace\_work\1\s\artifacts\logs\UnitTests-netcoreapp3.0-20190405.90_2019-04-06_01-21-58-692.trx
  
  Total tests: 23063. Passed: 23008. Failed: 1. Skipped: 54.
```

This is also hitting https://github.com/aspnet/AspNetCore/pull/9140 so I don't think it's specific to the changes we made and more likely a systemic issue?

**Comment by dougbu** [comment]
This isn't familiar to me.

Have you asked RaaS to triage the build failure or searched AspNetCore and AspNetCore-Internal for the relevant test name?

**Comment by dougbu** [comment]
On second thought, the failing test is also failing in internal builds. I'll add you to an internal thread&hellip;

**Comment by JunTaoLuo** [comment]
Retriggering tests since retries don't work.

**Comment by JunTaoLuo** [comment]
This covered: https://github.com/grpc/grpc-dotnet/issues/191 https://github.com/aspnet/AspNetCore/issues/7973 https://github.com/grpc/grpc-dotnet/issues/127 https://github.com/aspnet/AspNetCore/issues/7835
